### ISSUE TITLE
HTMLHELP: Underscore in chm file name prevents linking between chm files (Origin: bugzilla #334859)

### ISSUE THREAD

**Issue by doxygen** [doxygen/doxygen#2079]
status NEW severity *minor* in component *general* for *---*
Reported in version *1.4.x* on platform *Other*
Assigned to: Dimitri van Heesch

On 2006-03-17 10:59:50 +0000, Ernst Schwab wrote:

    Please describe the problem:
    When linking between several CHM files, as decribed in FAQ item 6, incorrect
    links are created when the CHM file contains an underscore character.
    
    
    Steps to reproduce:
    1. TAGFILES = dmx_com.tag=dmx_com.chm::
    2. TAGFILES = dmx_com.tag=dmx-com.chm::
    
    
    Actual results:
    In case 1, a link does not work and looks like this: 
    file:///C:/tmp/html/dmx_com.chm::/index.html#dmx_com_mainpage
    
    In case 2, the link looks like this and is correct:
    dmx-com.chm::/index.html#dmx_com_mainpage
    
    Expected results:
    
    
    Does this happen every time?
    Yes, reproducable with Windows version "Doxygen version 1.4.6-NO"
    
    Other information:

**Comment by albert-github** [comment]
The problem here looks like a bit of a quirk in the html help compiler.
It can be corrected by using:
```
TAGFILES = dmx_com.tag=mk:@MSITStore:dmx_com.chm::
```
instead of 
```
TAGFILES = dmx_com.tag=dmx_com.chm::
```

Using the `mk:@MSITStore:` in front of `dmx-com.chm::` does also work.

I think the problem is usage, though in the mean time I've just pushed a proposed documentation patch, pull request #9248

@doxygen can we mention this just in the documentation or should we try to fix this in the `htmlhelp.cpp` or `tagreader.cpp`?

**Comment by doxygen** [comment]
@albert-github If adding the `mk:@MSITStore:` prefix works in general we could also put this as a generic guideline in the manual for all references to chm files.

**Comment by albert-github** [comment]
@doxygen as far as I know it is valid for all chm files. I'll change the FAQ documentation (tomorrow).

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
